### DISCUSSION TITLE
dotnet6 docker build create a none images

### DISCUSSION THREAD

**Discussion by a63245111** [dotnet/dotnet-docker#3383] [General]
### Describe the Bug
In contos 7, i build my images. I found that an empty images was created. Dockerfile generated by vs 2022

```
my images
PEPOSITORY                     TAG          IMAGE ID           CREATED         SIZE
<none>                            <none>    38885d736caf   8 days ago    1.35GB
<none>                            <none>    5f08df1b82e1   8 days ago    248MB
<none>                            <none>    8bd9a012cf60   8 days ago    1.35GB
api                                        latest    49614d4398c5   10 days ago   255MB
<none>                            <none>    91dde5298666   10 days ago   1.51GB
<none>                            <none>    7594cdd25dfe   11 days ago   255MB
<none>                            <none>    89f995088619   11 days ago   1.51GB
<none>                            <none>    68db2020c1f5   12 days ago   255MB
<none>                            <none>    e0439b5ec9e1   12 days ago   1.51GB
<none>                            <none>    16533f151a3c   3 weeks ago   248MB
mcr.microsoft.com/dotnet/sdk      6.0       c8231459539b   5 weeks ago   715MB
mcr.microsoft.com/dotnet/aspnet   6.0       e38d4d116f1a   5 weeks ago   207MB
```

### Dockerfile

```Dockerfile
FROM mcr.microsoft.com/dotnet/aspnet:6.0 AS base
WORKDIR /app
EXPOSE 44307

FROM mcr.microsoft.com/dotnet/sdk:6.0 AS build
WORKDIR /src
COPY ["NuGet.Config", "."]
COPY ["src/ToPayManage.HttpApi.Host/ToPayManage.HttpApi.Host.csproj", "src/ToPayManage.HttpApi.Host/"]
COPY ["src/ToPayManage.HttpApi/ToPayManage.HttpApi.csproj", "src/ToPayManage.HttpApi/"]
COPY ["src/ToPayManage.Application.Contracts/ToPayManage.Application.Contracts.csproj", "src/ToPayManage.Application.Contracts/"]
COPY ["src/ToPayManage.Domain.Shared/ToPayManage.Domain.Shared.csproj", "src/ToPayManage.Domain.Shared/"]
COPY ["src/ToPayManage.EntityFrameworkCore/ToPayManage.EntityFrameworkCore.csproj", "src/ToPayManage.EntityFrameworkCore/"]
COPY ["src/ToPayManage.Domain/ToPayManage.Domain.csproj", "src/ToPayManage.Domain/"]
COPY ["src/ToPayManage.Application/ToPayManage.Application.csproj", "src/ToPayManage.Application/"]
COPY ["src/ToPayManage.Common/ToPayManage.Common.csproj", "src/ToPayManage.Common/"]
RUN dotnet restore "src/ToPayManage.HttpApi.Host/ToPayManage.HttpApi.Host.csproj"
COPY . .
WORKDIR "/src/src/ToPayManage.HttpApi.Host"
RUN dotnet build "ToPayManage.HttpApi.Host.csproj" -c Release -o /app/build

FROM build AS publish
RUN dotnet publish "ToPayManage.HttpApi.Host.csproj" -c Release -o /app/publish

FROM base AS final
WORKDIR /app
COPY --from=publish /app/publish .
ENV ASPNETCORE_URLS=http://+:44307

ENTRYPOINT ["dotnet", "ToPayManage.HttpApi.Host.dll"]
```

### Output of `docker version`

 Version:          20.10.11

### Output of `docker info`

client:
 Context:    default
 Debug Mode: false
 Plugins:
  app: Docker App (Docker Inc., v0.9.1-beta3)
  buildx: Build with BuildKit (Docker Inc., v0.6.3-docker)
  scan: Docker Scan (Docker Inc., v0.9.0)

Server:
 Containers: 2
  Running: 2
  Paused: 0
  Stopped: 0
 Images: 65
 Server Version: 20.10.11
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 1
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: runc io.containerd.runc.v2 io.containerd.runtime.v1.linux
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 7b11cfaabd73bb80907dd23182b9347b4245eb5d
 runc version: v1.0.2-0-g52b36a2
 init version: de40ad0
 Security Options:
  seccomp
   Profile: default
 Kernel Version: 3.10.0-1160.15.2.el7.x86_64
 Operating System: CentOS Linux 7 (Core)
 OSType: linux
 Architecture: x86_64
 CPUs: 2
 Total Memory: 3.7GiB
 Name: hecs-38405
 ID: D4OP:ZVDQ:SSVW:FJP3:7RIQ:JRFC:7O66:ITXU:AIL2:H7EX:ECKN:UVQX
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false

**Comment by mthalman** [comment]
Can you provide more information about your issue? What does your Dockerfile look like? What command is being run to build it? Are there any error messages?

> I found that an empty images was created.

What do you mean by "empty"?

**Comment by a63245111** [comment]
ok . i submitted.   it's a none  images  ,but it has about  2G size

**Comment by mthalman** [comment]
You're referring to the `<none>` that is showing up in the [output of `docker images`](https://docs.docker.com/engine/reference/commandline/images/)? That is completely normal, an implementation of Docker itself, and not related to the .NET Docker images. It just means that Docker's cached images contains an untagged image.

Is there a particular issue you are experiencing with the .NET Docker images? Are you not able to build or execute the container successfully? Please provide more info.

**Comment by a63245111** [comment]
emmm ,  Has been successfully built .  but  after build create two images.  the  none images more then 1 G size.

**Comment by mthalman** [comment]
Please be very explicit with what you see as the problem here. Is it strictly the size of the images that are in your Docker cache that you're concerned about?

Be aware that these are intermediate images that are stored by Docker as part of the build and do not indicate the image size of the final image of your application. Those intermediate images are a consequence of the docker build implementation and not specific to .NET images.

To see the final image size of your application, you can run the following command to get the number of bytes: `docker inspect --format "{{ .Size }}" <tag>`.
